File: HeartForCharity/HeartForCharity.Services/Database/Notification.cs
using HeartForCharity.Model.Enums;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HeartForCharity.Services.Database
{
    public class Notification
    {
        [Key]
        public int NotificationId { get; set; }

        [Required]
        [ForeignKey(nameof(UserProfile))]
        public int UserProfileId { get; set; }  

        [ForeignKey(nameof(VolunteerApplication))]
        public int? VolunteerApplicationId { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = null!;

        [Required]
        [MaxLength(2000)]
        public string Message { get; set; } = null!;

        public NotificationType Type { get; set; } = NotificationType.General;
        public bool IsRead { get; set; } = false;

        public DateTime SentDateTime { get; set; } = DateTime.UtcNow;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigacijska svojstva
        public virtual UserProfile UserProfile { get; set; } = null!;
        public virtual VolunteerApplication? VolunteerApplication { get; set; }
    }
}

File: HeartForCharity/UI/heartforcharity_mobile/lib/model/responses/notification.dart
class AppNotification {
  final int notificationId;
  final int userProfileId;
  final int? volunteerApplicationId;
  final String title;
  final String message;
  final String type;
  bool isRead;
  final DateTime sentDateTime;

  AppNotification({
    this.notificationId = 0,
    this.userProfileId = 0,
    this.volunteerApplicationId,
    this.title = '',
    this.message = '',
    this.type = 'General',
    this.isRead = false,
    DateTime? sentDateTime,
  }) : sentDateTime = sentDateTime ?? DateTime.now();

  factory AppNotification.fromJson(Map<String, dynamic> json) => AppNotification(
        notificationId: json['notificationId'] ?? 0,
        userProfileId: json['userProfileId'] ?? 0,
        volunteerApplicationId: json['volunteerApplicationId'],
        title: json['title'] ?? '',
        message: json['message'] ?? '',
        type: json['type'] ?? 'General',
        isRead: json['isRead'] ?? false,
        sentDateTime: json['sentDateTime'] != null
            ? DateTime.parse(json['sentDateTime'])
            : DateTime.now(),
      );
}
